### ISSUE TITLE
import/no-extraneous-dependencies configuration options are not honored – Reproducer included!

### ISSUE THREAD

**Issue by albertodiazdorado** [microsoft/vscode-eslint#1783]
If I configure my `.eslintrc.js` like this, then VS Code displays false positives. That means: whilst `npx eslint *` is green, VS Code reports errors:
```js
module.exports = {
  root: true,
    extends: [
      "plugin:import/recommended",
    ],
    rules: {
      "import/no-extraneous-dependencies": [
        "error",
        { devDependencies: ["test/**/*.mjs"] },
      ],
    },
  };
```
![image](https://github.com/microsoft/vscode-eslint/assets/16071626/bf2a5051-c497-400e-8022-59e83e813155)

If I change the glob pattern as follows, then there are no more false positives:
```js
module.exports = {
  root: true,
    extends: [
      "plugin:import/recommended",
    ],
    rules: {
      "import/no-extraneous-dependencies": [
        "error",
        /{ devDependencies: ["**/*.test.mjs"] },
      ],
    },
  };
```
![image](https://github.com/microsoft/vscode-eslint/assets/16071626/cc932059-bfa2-47ad-a53d-aa8a966eb0d3)

Reproducer: https://github.com/albertodiazdorado/eslint-bug-demo

**Comment by albertodiazdorado** [comment]
Nice. Thanks!
